app/controllers/task_controller.rb

```diff
@@ -37,10 +37,40 @@ def confirm_delete
     redirect_to :root
   end
 
+  def update
+    Task.update((params[:id]), completed_on: Time.now)
+
+    index
+  end
+
+  def uncheck_update
+    Task.update((params[:id]), completed_on:" ")
+
+    index
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+
+    render :edit_task
+  end
+
+  def edit_update
+    @task = Task.find(params[:id])
+    @name = create_params[:task][:name]
```

Comment:
It seems like you are using unnecessary instance variables here? Was there a specific reason you didn't use `create_params[:task]` and instead used individual variables?